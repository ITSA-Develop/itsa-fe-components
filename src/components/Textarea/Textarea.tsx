import { Input } from 'antd';
import { TextAreaProps } from 'antd/es/input';

const { TextArea: AntTextArea } = Input;

export const Textarea = ({ ...rest }: TextAreaProps) => {
	return <AntTextArea {...rest} />;
};
