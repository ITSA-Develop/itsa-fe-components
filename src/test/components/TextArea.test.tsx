import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { describe, expect, it } from 'vitest';
import { Textarea } from '../../components/Textarea/Textarea';

describe('Textarea component', () => {
	it('renders with given props and allows typing', async () => {
		const { container } = render(<Textarea placeholder="Escribe algo..." data-testid="textarea" />);
		const textarea = screen.getByTestId('textarea');

		expect(textarea).toBeInTheDocument();
		expect(textarea).toHaveAttribute('placeholder', 'Escribe algo...');

		await userEvent.type(textarea, 'Hola mundo');
		expect(textarea).toHaveValue('Hola mundo');
		expect(container).toMatchSnapshot();
	});

	it('accepts and passes extra props', () => {
		const { container } = render(<Textarea maxLength={10} data-testid="textarea" />);
		const textarea = screen.getByTestId('textarea');
		expect(textarea).toHaveAttribute('maxlength', '10');
		expect(container).toMatchSnapshot();
	});
});
