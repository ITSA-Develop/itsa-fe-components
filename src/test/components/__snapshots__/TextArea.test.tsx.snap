// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Textarea component > accepts and passes extra props 1`] = `
<div>
  <textarea
    class="ant-input css-dev-only-do-not-override-l9pxc0 ant-input-outlined"
    data-testid="textarea"
    maxlength="10"
  />
</div>
`;

exports[`Textarea component > renders with given props and allows typing 1`] = `
<div>
  <textarea
    class="ant-input css-dev-only-do-not-override-l9pxc0 ant-input-outlined"
    data-testid="textarea"
    placeholder="Escribe algo..."
  >
    Hola mundo
  </textarea>
</div>
`;
